app/src/main/java/io/gnosis/safe/ui/settings/owner/OwnerSeedPhraseViewModel.kt

```diff
@@ -12,7 +12,42 @@ class OwnerSeedPhraseViewModel
     appDispatchers: AppDispatchers
 ) : PublishViewModel<ImportOwnerKeyState>(appDispatchers) {
 
-    fun validate(seedPhrase: String) {
+    fun validate(seedPhraseOrKey: String) {
+        if (isPrivateKey(seedPhraseOrKey)) {
+            validatePrivateKey(seedPhraseOrKey)
+        } else {
+            validateSeedPhrase(seedPhraseOrKey)
+        }
+    }
+
+    internal fun isPrivateKey(seedPhraseOrKey: String): Boolean {
+        val input = removeHexPrefix(seedPhraseOrKey)
+
+        val pattern = "[0-9a-fA-F]{64}".toRegex()
+        pattern.matchEntire(input)?.let {
+            return true
+        }
+
+        return false
+    }
+
+    internal fun validatePrivateKey(key: String) {
```

Comment:
do we need to check if input is private key here or can we just assume it? validatePrivateKey should be called after isPrivateKey

Reply:
I added this check so we could use validatePrivateKey in other contexts. But I can remove it. It is not necessary as we use it right now.

Reply:
changed in: 9d9a67cfec986d158d5763a4d4392c302ec70e3a